src/fundamental/uki.c

```diff
@@ -9,19 +9,20 @@ const char* const unified_sections[_UNIFIED_SECTION_MAX + 1] = {
          * https://learn.microsoft.com/en-us/windows/win32/debug/pe-format#section-table-section-headers
          * (Note that PE *object* files may have longer section names (via indirection in the string table) but
          * this is not allowed for PE *executables*, which UKIs are.) */
-        [UNIFIED_SECTION_LINUX]   = ".linux",
-        [UNIFIED_SECTION_OSREL]   = ".osrel",
-        [UNIFIED_SECTION_CMDLINE] = ".cmdline",
-        [UNIFIED_SECTION_INITRD]  = ".initrd",
-        [UNIFIED_SECTION_UCODE]   = ".ucode",
-        [UNIFIED_SECTION_SPLASH]  = ".splash",
-        [UNIFIED_SECTION_DTB]     = ".dtb",
-        [UNIFIED_SECTION_UNAME]   = ".uname",
-        [UNIFIED_SECTION_SBAT]    = ".sbat",
-        [UNIFIED_SECTION_PCRSIG]  = ".pcrsig",
-        [UNIFIED_SECTION_PCRPKEY] = ".pcrpkey",
-        [UNIFIED_SECTION_PROFILE] = ".profile",
-        [UNIFIED_SECTION_DTBAUTO] = ".dtbauto",
-        [UNIFIED_SECTION_HWIDS]   = ".hwids",
+        [UNIFIED_SECTION_LINUX]      = ".linux",
+        [UNIFIED_SECTION_OSREL]      = ".osrel",
+        [UNIFIED_SECTION_CMDLINE]    = ".cmdline",
+        [UNIFIED_SECTION_INITRD]     = ".initrd",
+        [UNIFIED_SECTION_UCODE]      = ".ucode",
+        [UNIFIED_SECTION_SPLASH]     = ".splash",
+        [UNIFIED_SECTION_DTB]        = ".dtb",
+        [UNIFIED_SECTION_UNAME]      = ".uname",
+        [UNIFIED_SECTION_SBAT]       = ".sbat",
+        [UNIFIED_SECTION_PCRSIG]     = ".pcrsig",
+        [UNIFIED_SECTION_PCRPKEY]    = ".pcrpkey",
+        [UNIFIED_SECTION_PROFILE]    = ".profile",
+        [UNIFIED_SECTION_DTBAUTO]    = ".dtbauto",
+        [UNIFIED_SECTION_HWIDS]      = ".hwids",
+        [UNIFIED_SECTION_FIRMWARE]   = ".efifwauto",
```

Comment:
As per the comment on the 8th line of this file, ".efifwauto" is too long.

Reply:
yes will fix this.

Reply:
fixed. I dropped "efi" since I do not see any reason why we should have a separate section for "efi" firmwares and a separate one for other types of firmwares. The "compatible" attribute will only select one, the first matching one for a given hw.